### ISSUE TITLE
Add support for SUPABASE_PROJECT_ID env variable in place of --project-ref flag

### ISSUE THREAD

**Issue by martypdx** [supabase/cli#1627]
**Describe the bug**
According to the [docs](https://supabase.com/docs/guides/cli/managing-environments#configure-github-actions), the cli should be able to use the following env vars:

- SUPABASE_ACCESS_TOKEN is your personal access token
- SUPABASE_DB_PASSWORD is your project specific database password
- SUPABASE_PROJECT_ID is your project specific reference string

The first two work fine, but cli will fail if `--project-ref <project-id>` is not included. Tried SUPABASE_PROJECT_ID, PROJECT_ID, SUPABASE_PROJECT_REF

**To Reproduce**
1. Create an `.env` file or other means to pass env variables to a supabase cli command
2. Supabase should not be linked
3. Run a command that needs `--project-ref`
5. Fails with error like "Cannot find project ref. Have you run supabase link?"

**Expected behavior**
The cli uses the provided `SUPABASE_PROJECT_ID` env variable value when attempting the supabase cli command

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version of CLI: 1.106.1
 - Version of Node.js: 21.1

**Comment by kamilogorek** [comment]
This is somewhat of a wording issue in the docs. If you see the usage example, it's explicitly using those variables, that's why docs state you need to set them first before using examples, see: https://github.com/supabase/supabase-action-example/blob/d201824df60e6ee22c8d72788296e5cb919a2ee5/.github/workflows/production.yaml#L25C45-L25C45

CLI is not using them right now, but I agree it would be a nice touch.

**Comment by martypdx** [comment]
I opened a [PR on docs](https://github.com/supabase/supabase/pull/18735) for this.

@kamilogorek The issue was never about the examples or whether I was able to get it to work. Call me old-fashion, but getting it to work on my machine is hardly the end of the sw dev process. Building out and refactoring the full pipeline for the project, it added a few extra hours spent hours tracking this down amongst the other factors like GH secrets or vars.
